src/createTheme.ts

```diff
@@ -0,0 +1,193 @@
+import { DefaultTheme } from 'styled-components';
+import { ThemeVariants } from './types/ThemeVariants';
+import { defaultThemeColors } from './defaultThemeColors';
+import { ThemeBreakpoints } from './types/ThemeBreakpoints';
+import { ButtonVariants } from './types/ButtonVariants';
+import { BadgeVariants } from './types/BadgeVariants';
+import { defaultThemeSizes } from './defaultThemeSizes';
```

Comment:
I can see what you're trying to do here but there is a cleaner way of achieving stronger type safety by using Typescript's `as const` syntax. We can avoid all these intermediate variant types by marking the style object `as const`:

```tsx
export const createTheme = () => {
  const colors = {
    primaryLightest: '#CADCFF',
    primaryLight: '#226EFF',
    primary: '#0958F3',
    primaryDark: '#0046CE',
  } as const;

  const buttonVariants = {
    primary: {
      backgroundColor: colors.primary,
      color: colors.white,
      '&:hover': {
        backgroundColor: colors.primaryLight,
      },
      '&:active': {
        backgroundColor: colors.primaryDark,
      },
      '&:disabled': {
        backgroundColor: colors.primaryLightest,
      },
    },
  } as const;

  return {
    colors,
    space: [0, 4, 8, 16, 24, 32, 64, 126, 256] as const,
    typography: {
      fontSize: 16,
      color: colors.black,
      bodyFontFamily: 'Avenir',
      headerFontFamily: 'Tiempos',
    } as const,

    // ...etc
  };
};

```

<img width="887" alt="Screen Shot 2020-05-22 at 12 06 28 PM" src="https://user-images.githubusercontent.com/6404663/82701271-b523cf00-9c24-11ea-93f4-e6d4726bb187.png">

Reply:
neat! let me update to use `const`

Reply:
🤔 Does that mean we can only ever have one default and the type is strictly based on the default?

Reply:
@vietnogigoodrx can you give me an example of you mean?

Reply:
While implementing the `as const`, I started feeling that we should have explicit types defined.    I think it's easier to understand the types when they're explicit.  I am also afraid that changes to the default value may have unintentional changes to the type.  I was planning to export these types so that the web-app, so I think its a good idea to be explicit to reduce the chances of breakage.

Reply:
@vietnogigoodrx I'm not sure what you mean by explicit types. 

A type of 
```ts
type Colors = {
  default: 'some-hex-literal';
}
```

is more explicit than 

```ts
type Colors = {
  default: string;
}
```

With the former it'll be very easy in the ID to see what hex color is being used and with the latter, all you'll see for the type is "string".

Reply:
To make sure we're on the same page, I want to make sure the original comment was suggesting:

```
const defaultColors = {
  default: '#494D55',
  black: '#1B202B',
} as const;

type ThemeColors = typeof defaultColors
```

instead of

```
interface ThemeColors {
  default: string;
  black: string;
}

const defaultColors: ThemeColors = {
  default: '#494D55',
  black: '#1B202B',
}
```

Reply:
@vietnogigoodrx I created a [Codesandbox](https://codesandbox.io/s/styled-components-typescript-ntpgg?file=/src/App.tsx) to demonstrate what I mean 😄

Reply:
Cool, I took a look and added couple of lines (`const myCustomTheme: Theme = {`) to illustrate a problem.

Reply:
@erikshestopal👆🏻

Reply:
@vietnogigoodrx You're gonna have fork the sandbox and post another link because I can't see your changes.

Reply:
@erikshestopal can you see the code changes in this link?  https://codesandbox.io/s/styled-components-typescript-csdby?file=/src/App.tsx

Reply:
@vietnogigoodrx Here's how I would approach that problem: https://codesandbox.io/s/styled-components-typescript-46y3w?file=/src/App.tsx

Reply:
@erikshestopal  That solution does not yield the same result.  I wanted to created another `Theme` type not a `DefaultTheme` type.

Reply:
I can understand overriding the default `molekule` theme, say in the EMR where we have different colors, but why would we have a second theme? My proposed solution provides the ability to override the default theme per app if we need different values.

Reply:
If I were to override the theme, I would use the `Theme` type to tell me which keys I can override and what type it should be.  In the proposed solution, 

```
const customTheme: DefaultTheme = {
  ...theme,
  colors: {
    ...theme.colors,
    primaryLight: "white",
    primary: 123,
    randomeKey: 'foo',
  }
};
```
has no type checking.

Reply:
@erikshestopal ☝🏻

Reply:
@vietnogigoodrx I'm not sure I follow because I'm getting the typechecking that you're looking for 🤔 

<img width="481" alt="Screen Shot 2020-05-27 at 2 08 13 PM" src="https://user-images.githubusercontent.com/6404663/83072401-8f7f3700-a023-11ea-8fb1-5ba23829be4e.png">

Reply:
@erikshestopal hmm, thats odd.  `DeefaultTheme` is pretty much any object so it shouldn't error.  This is whats showing up for me:

![typing](https://user-images.githubusercontent.com/58744872/83077061-8befae00-a02b-11ea-967c-c53e630ac13e.gif)